manage_breast_screening/notifications/management/commands/retry_failed_message_batch.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import time
+from logging import getLogger
+
+from django.core.management.base import BaseCommand, CommandError
+
+from manage_breast_screening.notifications.management.commands.command_helpers import (
+    MessageBatchHelpers,
+)
+from manage_breast_screening.notifications.models import (
+    MessageBatch,
+    MessageBatchStatusChoices,
+)
+from manage_breast_screening.notifications.services.api_client import ApiClient
+from manage_breast_screening.notifications.services.queue import Queue
+
+logger = getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django Admin command which takes an ID of a MessageBatch with
+    a failed status and retries sending it to the Communications API.
+    """
+
+    def handle(self, *args, **options):
+        queue = Queue.RetryMessageBatches()
+        queue_message = queue.item()
+
+        if queue_message is None:
+            logger.info("No messages on queue")
+            return
+
+        message_batch_id = json.loads(queue_message.content)["message_batch_id"]
+        message_batch = MessageBatch.objects.filter(
+            id=message_batch_id,
+            status=MessageBatchStatusChoices.FAILED_RECOVERABLE.value,
+        ).first()
+
+        if message_batch is None:
+            raise CommandError(
+                f"Message Batch with id {message_batch_id} and status of '{MessageBatchStatusChoices.FAILED_RECOVERABLE.value}' not found"
+            )
+
+        queue.delete(queue_message)
+
+        retry_count = int(json.loads(queue_message.content)["retry_count"])
+        if retry_count < int(os.getenv("RETRY_LIMIT", "5")):
+            time.sleep(int(os.getenv("RETRY_DELAY", "0")) * retry_count)
+
+            try:
+                response = ApiClient().send_message_batch(message_batch)
+
+                if response.status_code == 201:
```

Comment:
Would we want to apply the same logic as we do here https://github.com/NHSDigital/dtos-manage-breast-screening/blob/main/manage_breast_screening/notifications/management/commands/command_helpers.py#L52-L64 in determining what to do with the response?
Asking because a batch might be added to the retry queue then retried and fail for a completely different reason.
It feels like the response handling logic should be shared somehow.